compiler/deSugar/Match.hs

```diff
@@ -161,18 +161,19 @@ type MatchId = Id   -- See Note [Match Ids]
 match :: [MatchId]        -- Variables rep\'ing the exprs we\'re matching with
                           -- See Note [Match Ids]
       -> Type             -- Type of the case expression
+      -> Rig              -- Weight of the case expression
```

Comment:
Lets say « Multiplicity » instead of « Weight » here. (we will change the names of the modules and types later, but let's get the comments aligned with the paper and proposal).